specs/HostResourceMapping.md

```diff
@@ -0,0 +1,108 @@
+# Background
+We have received feedbacks that there is a need to access local files in WebView2. In this document we describe the APIs for this scenario. We'd appreciate your feedback.
+
+
+# Description
+To access local files in WebView2, define a virutal host name to local folder mapping using the `AddVirtualHostNameToFolderMapping` API,
+and then refer to local files under the folder with a normal http/https url with the virtual host name.
+
+# Examples
+Suppose the app has index.html file next to its exe file. The following code snippet demonstrates how to define a mapping
+for `app-file.invalid` host name so that it can be accessed from pages using normal http/https url like https://app-file.invalid/index.html.
+
+## Win32 C++
+```cpp
+webview2->AddVirtualHostNameToFolderMapping(
+    L"app-file.invalid", L".", COREWEBVIEW2_HOST_RESOURCE_ACCESS_KIND_DENY_CORS);
```

Comment:
Samples should demonstrate recommended security best practices by isolating the files to be shared into a subdirectory and sharing the subdirectory. Sharing the entire application directory means that you're sharing far more than just the index file; you're also sharing the executable and other supporting files in the executable directory.

Reply:
Good point. This could be L".\\assets" instead of L"."

Reply:
Does the path use "\\" or "/" or either?

Reply:
Good question. It will be nice to support both since we eventually will run on Mac & Linux.

Reply:
Updated and simply uses "assets". It is file path and could be different on Windows and Mac/Linux.